guides/request-validation-csharp/example-2/example-2.4.x.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Configuration;
+using System.Web;
+using System.Web.Mvc;
+using Twilio.TwiML;
+using Twilio.TwiML.Mvc;
+
+namespace helloworldcsharp.Controllers
+{
+	public class IncomingController : Controller
+	{
+		[RequestValidatorFilter]
+		public ActionResult Voice (string From)
+		{
+			var response = new TwilioResponse();
+			var message = @"
+				Thanks for calling!
+
+    			Your phone number is {0}. I got your call because of Twilio's webhook.
+
+    			Goodbye!
+			";
+			response.Say(string.Format(message, From));
+			response.Hangup();
+
+			return new TwiMLResult(response);
```

Comment:
The reason to using `TwilioController` as the base class for this controller is to use the `TwiML` helper, which is just a shortcut to what you did in this line.

```charp
return TwiML(response);
```